### PR TITLE
Add 0.16 blocks/items

### PR DESCRIPTION
**Pull request by buchwasa**
This PR adds all (most if I'm wrong) of 0.16 blocks and items.

### CODE REVIEW

File: src/pocketmine/resources/creativeitems.json [hunk 1]
```diff
@@ -593,6 +593,24 @@
         "count": 1,
         "nbt": ""
     },
+	{
```
Comment (dktapps) [inline src/pocketmine/resources/creativeitems.json]: This file is auto-generated, don't modify it by hand. It should already be up-to-date for 0.16.

File: src/pocketmine/item/ItemIds.php [hunk 2]
```diff
@@ -176,7 +176,9 @@
 	const QUARTZ = 406;
 	const NETHER_QUARTZ = 406;
 	const MINECART_WITH_TNT = 407;
-	const MINECART_WITH_HOPPER = 408, HOPPER = 410;
```
Comment (dktapps) [inline src/pocketmine/item/ItemIds.php]: lol @SOF3 , blame your automated formatting ;P

File: src/pocketmine/block/Beacon.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	
+ 	public function place(Item $item, Block $block, Block $target, $face, $fx, $fy, $fz, Player $player = null){
+ 		$this->getLevel()->setBlock($this, $this, true, true);
+ 		$nbt = new CompoundTag("", [
+ 			new StringTag("id", Tile::BEACON),
+ 			new IntTag("x", $block->x),
+ 			new IntTag("y", $block->y),
+ 			new IntTag("z", $block->z)
+ 		]);
+ 		$pot = Tile::createTile(Tile::BEACON, $this->getLevel()->getChunk($this->x >> 4, $this->z >> 4), $nbt);
```
Comment (dktapps) [inline src/pocketmine/block/Beacon.php]: careful with the copy-paste ;)

File: src/pocketmine/block/Prismarine.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+	public function __construct($meta = 0){
+		$this->meta = $meta;
+	}
+	
+	public function getHardness(){
+		return 1.5;
+	}
+
+	public function getName() : string{
```
Comment (dktapps) [inline src/pocketmine/block/Prismarine.php]: iirc the base method doesn't have type-hint, you might want to remove this.

File: src/pocketmine/block/Prismarine.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+			self::BRICKS => "Prismarine Bricks",
+		];
+		return $names[$this->meta & 0x0f];
+	}
+
+	public function getToolType(){
+		return Tool::TYPE_PICKAXE;
+	}
+
+	public function getDrops(Item $item) : array {
```
Comment (dktapps) [inline src/pocketmine/block/Prismarine.php]: same here

File: src/pocketmine/block/SeaLantern.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+class SeaLantern extends Solid{
+
+	protected $id = self::SEA_LANTERN;
+
+	public function __construct($meta = 0){
+		$this->meta = $meta;
+	}
+
+	public function getName() : string{
```
Comment (dktapps) [inline src/pocketmine/block/SeaLantern.php]: and here.. I think you get the idea

**Review by dktapps — CHANGES_REQUESTED** [review]
BTW the reason we haven't added these ourselves is because we are currently planning rewrites.

Please attend to my comments on the diff.

**Comment by AppleDevelops** [comment]
I'll work on it when I get back. It's copy and pasted from my dead software which used a lot of spaces. (Everyone used GitHub web editor for some reason)

File: src/pocketmine/tile/Beacon.php [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+ use pocketmine\level\format\Chunk;
+ use pocketmine\nbt\tag\CompoundTag;
+ use pocketmine\nbt\tag\IntTag;
+ use pocketmine\nbt\tag\StringTag;
+ use pocketmine\nbt\tag\ByteTag;
+ use pocketmine\nbt\tag\FloatTag;
+ 
+ class Beacon extends Spawnable{
+ 
+ 	public function __construct(Chunk $chunk, CompoundTag $nbt){
```
Comment (dktapps) [inline src/pocketmine/tile/Beacon.php]: You're PRing to master, which doesn't have 1.0 changes yet. This should more correctly be FullChunk on the current master.

Comment (AppleDevelops) [inline src/pocketmine/tile/Beacon.php]: @dktapps My bad 😄

**Comment by AppleDevelops** [comment]
There you go :)

**Review by dktapps — APPROVED** [review]

**Comment by sylvrs** [comment]
finally, 0.16 support

**Pull request merged**